validator/beacon/service.go

```diff
@@ -173,55 +160,43 @@ func (s *Service) listenForAssignmentChange(client pb.BeaconServiceClient) {
 			break
 		}
 
-		for _, assign := range assignment.Assignments {
-			if bytes.Equal(assign.PublicKey.PublicKey, s.pubKey) {
-				s.role = assign.Role
-				if s.CurrentCycleStartSlot(params.DemoConfig().CycleLength) == 0 {
-					// +1 to account for genesis block being slot 0.
-					s.assignedSlot = params.DemoConfig().CycleLength + assign.AssignedSlot + 1
-				} else {
-					s.assignedSlot = s.CurrentCycleStartSlot(params.DemoConfig().CycleLength) + assign.AssignedSlot + 1
```

Comment:
Assuming a `cycleLength = 5` and `assignedSlot = 0`,  this would set `s.assignedSlot` to 1, 6, 11, etc. instead of 0, 5, 10, etc.